kikit/panelize.py

```diff
@@ -1104,7 +1105,12 @@ def makeGrid(self, boardfile: str, sourceArea: BOX2I, rows: int, cols: int,
 
         Parameters:
 
-        boardfile - the path to the filename of the board to be added
+        boardfile - the path to the filename of the board to be added.
+            If a file with `-rX-cY` appended to boardfile, where X and Y are the
+            row and column number of that position on the grid respectively,
+            is found, then `makeGrid()` will use that file in that position of
+            the grid instead. All boards must be compatible: have the same
+            board outline, stack up, DRC rules, etc.
```

Comment:
I don't particularly like the naming scheme, as it is fixed. To be honest, it seems more like a quick'n'dirty hack for your particular use case (that can be handled probably better by writing a layout plugin).

Instead, we should extend `makeGrid` by a `boardProvider` that delivers the source boards for makeGrid. The `boardProvider` is a point of customizability. W can then, e.g., add command line options that will take boards based on the specified naming schemes or where we provide a list of source boards.

Similarly, in the spirit of "aborting on dangerous operations," your code never checks if the boards are of the same size. And it can produce faulty results on size mismatch. We can probably load all the boards, find the largest one and use its size for grid cell size.